src/Entity.js

```diff
@@ -214,22 +259,9 @@ exports = Class(function() {
 	 *         ~ returns the total distance moved by the two entities
 	 */
 
-	this.collidesWith = function(entity) {
-		return this.physics.collide(this, entity);
-	};
-
-	this.resolveCollidingStateWith = function(entity) {
```

Comment:
@tuanna222 SATPhysics should also support this API by using `response.overlap` as you mentioned. Can you change the call to `resolveCollidingStateWith` to accept an optional `response` object, which gets passed into physics? You'll need to add the missing function `resolveCollidingState` to SATPhysics. The other missing functions `resolveCollidingCircles`, `resolveCollidingCircleRect`, and `resolveCollidingRects` are specific to EntityPhysics; I imagine the SATPhysics counterparts might be similar, except replace Rect with Polygon?

In general, I marked the physics functions that are required by the Entity API with the tag `* ~ REQUIRED for Entity`.

I started a merger yesterday but have a lot of other stuff I'm working on. So far, I've repaired the README file and I'm working on removing most physics specific logic you added to Entity.js into SATPhysics and EntityPhysics. I'll push my code to the `merge-sat` branch, but it's still a work-in-progress, and not ready to merge. At least you can see the direction I was heading.

You will also want to match our style when you edit code in one of the original files. Adding external files in their own style is fine though, like SAT.js.